src/util/Logging.cpp

```diff
@@ -134,10 +134,20 @@ Logging::init(bool truncate)
             // and always pass 'truncate=false' to spdlog, so it always opens in
             // "ab" == O_APPEND mode. The "portable" way to truncate is to open
             // an ofstream in out|trunc mode, then immediately close it.
+            std::ofstream out;
             if (truncate)
             {
-                std::ofstream truncator(filename, std::ios_base::out |
-                                                      std::ios_base::trunc);
+                out.open(filename, std::ios_base::out | std::ios_base::trunc);
+            }
+            else
+            {
+                out.open(filename, std::ios_base::out | std::ios_base::app);
+            }
+
+            if (out.fail())
+            {
+                throw std::runtime_error(
```

Comment:
We should probably provide the path in the exception message

Reply:
updated